lib/internal/mime.js

```diff
@@ -182,103 +298,103 @@ class MIMEParams {
     for (const { 0: key, 1: value } of this.#data) {
       const encoded = encode(value);
       // Ensure they are separated
-      if (ret.length) ret += ';';
+      if (ret.length !== 0) ret += ';';
       ret += `${key}=${encoded}`;
     }
+
     return ret;
   }
 
   // Used to act as a friendly class to stringifying stuff
   // not meant to be exposed to users, could inject invalid values
   static parseParametersString(str, position, params) {
     const paramsMap = params.#data;
-    const endOfSource = SafeStringPrototypeSearch(
-      StringPrototypeSlice(str, position),
-      START_ENDING_WHITESPACE,
-    ) + position;
-    while (position < endOfSource) {
-      // Skip any whitespace before parameter
-      position += SafeStringPrototypeSearch(
-        StringPrototypeSlice(str, position),
-        END_BEGINNING_WHITESPACE,
-      );
-      // Read until ';' or '='
-      const afterParameterName = SafeStringPrototypeSearch(
-        StringPrototypeSlice(str, position),
-        EQUALS_SEMICOLON_OR_END,
-      ) + position;
-      const parameterString = toASCIILower(
-        StringPrototypeSlice(str, position, afterParameterName),
-      );
-      position = afterParameterName;
-      // If we found a terminating character
-      if (position < endOfSource) {
-        // Safe to use because we never do special actions for surrogate pairs
-        const char = StringPrototypeCharAt(str, position);
-        // Skip the terminating character
-        position += 1;
-        // Ignore parameters without values
-        if (char === ';') {
-          continue;
-        }
+    const endOfSource = str.length;
+    const marker = { position };
```

Comment:
is marker an object, because we want to keep it as byRef parameter?, needs comment